internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go

```diff
@@ -625,8 +618,7 @@ func (h *Handler) listSharesWithMe(w http.ResponseWriter, r *http.Request) {
 		var info *provider.ResourceInfo
 		if pinfo != nil {
 			// check if the shared resource matches the path resource
-			if rs.Share.ResourceId.StorageId != pinfo.GetId().StorageId ||
-				rs.Share.ResourceId.OpaqueId != pinfo.GetId().OpaqueId {
+			if !utils.ResourceEqual(&provider.Reference{ResourceId: rs.Share.ResourceId}, &provider.Reference{ResourceId: pinfo.Id}) {
```

Comment:
We can directly use `ResourceIDEqual` here and other places but those can be fixed later